workflows/src/main/kotlin/com/r3/corda/lib/tokens/workflows/internal/selection/DatabaseTokenSelection.kt

```diff
@@ -111,7 +114,7 @@ class DatabaseTokenSelection(
         val stateAndRefs = mutableListOf<StateAndRef<FungibleToken>>()
         for (retryCount in 1..maxRetries) {
             val additionalCriteria = if (queryBy?.issuer != null) {
-                tokenAmountWithIssuerCriteria(requiredAmount.token, queryBy.issuer)
+                tokenAmountWithIssuerCriteria(requiredAmount.token, queryBy.issuer!!) //TODO !!
```

Comment:
What's this bit ?

Reply:
ah, forgot to cleanup after refactor, good spot